library/cipher.c

```diff
@@ -311,6 +311,22 @@ int mbedtls_cipher_update( mbedtls_cipher_context_t *ctx, const unsigned char *i
         return MBEDTLS_ERR_CIPHER_INVALID_CONTEXT;
     }
 
+#if defined(MBEDTLS_CIPHER_MODE_CTR)
+    if( ctx->cipher_info->mode == MBEDTLS_MODE_CTR )
+    {
+        if( 0 != ( ret = ctx->cipher_info->base->ctr_func( ctx->cipher_ctx,
+                ilen, &ctx->unprocessed_len, ctx->iv,
+                ctx->unprocessed_data, input, output ) ) )
+        {
+            return( ret );
+        }
+
+        *olen = ilen;
+
+        return( 0 );
+    }
+#endif /* MBEDTLS_CIPHER_MODE_CTR */
+
     if( input == output &&
```

Comment:
why not add here a check that the mode is not CTR? It is much more cleaner to first do parameter checks, and then do the operations:
~~~
#if defined(MBEDTLS_CIPHER_MODE_CTR)
( ctx->cipher_info->mode != MBEDTLS_MODE_CTR ) &&
~~~

Reply:
I have seen that we already have ECB and GCM above the parameters check, therefore I thought that all the modes for which such check is not required should go above it.

But modifying the check to exclude CTR is also fine with me. Please let me know what you think is more meaningful.

Thanks!

Reply:
Hi @ordex, my personal view is modifying the check, but I would like to see what my colleagues have to say as well. Keep in mind that the change would probably be needed for CFB mode as well ( I haven't checked yet )

Reply:
Thanks for your feedback @RonEld . Let's wait for your colleagues before doing any change then.
Does also MODE_STREAM follow the same logic? I Am wondering if this check is valid only for CBC ...

Reply:
What is the current situation on this PR - is the PR included or is there other fixes for the bug #1076?